src/rhsmlib/file_monitor.py

```diff
@@ -0,0 +1,280 @@
+        loop function will override parent class loop function
+        :param dir_watches: list of directories to watch (see DirectoryWatch class below)
+        """
+        super(InotifyFilesystemWatcher, self).__init__(dir_watches)
+        self.watch_manager = None
+        self.notifier = None
+
+    def loop(self, callback=None):
+        """
+        sets up watch manager, notifier, adds watches to watch manager, and starts loop
+        :param callback: callback method to be called at the end of each iteration of the loop
+        :return:
+        """
+        self.watch_manager = pyinotify.WatchManager()
+        self.notifier = pyinotify.Notifier(self.watch_manager, self.handle_event)
+        self.add_watches()
+
+        def inotify_callback():  # arg: notifier, required for pyinotify
+            return end_loop_cb(self, user_callback=callback)
+
+        while not inotify_callback():
+            self.notifier.process_events()
+            if self.notifier.check_events(500):  # keeps tests reasonably fast while still timing out
+                self.notifier.read_events()
+
+    def handle_event(self, event):
+        """
+        default process function for pyinotify notifier
+        :param event: pyinotify Event object, has path and mask of flags representing file modification
+        :return:
+        """
+        for dir_watch in self.dir_watches:
+            if dir_watch.paths_match(event.path) and dir_watch.file_modified(event.mask):
+                dir_watch.notify()
+
+    def add_watches(self):
+        """
+        adds watches to the watch manager
+        :return:
+        """
+        for dir_watch in self.dir_watches:
+            self.watch_manager.add_watch(dir_watch.path, dir_watch.mask, do_glob=dir_watch.is_glob)
+
+
+class DirectoryWatch(object):
+    """
+    Directory to be watched
+
+    Included in list to be passed into filesystem watcher object
+    Example usage:
+        directory_watch = DirectoryWatch("~/home/", [my_function1, my_function2])
+    """
+    def __init__(self, path, callbacks, is_glob=False):
+        """
+        :param path: path associated with directory to be watched
+        :param callbacks: list of methods called when directory is changed
+        :param is_glob: bool - if path provided is glob or not
+        """
+        self.IN_CREATE = 0x00000100
+        self.IN_DELETE = 0x00000200
+        self.IN_MODIFY = 0x00000002
+
+        self.path = path
+        self.timestamp = None
+        self.is_glob = is_glob
+        self.callbacks = callbacks
+        self.mask = self.IN_CREATE | self.IN_DELETE | self.IN_MODIFY
+
+    def notify(self):
+        """
+        calls all callbacks associated with dir watch
+        :return:
+        """
+        for cb in self.callbacks:
+            try:
+                cb()
+            except Exception as e:
+                log.exception(e)
+
+    def changed(self, fsw):
```

Comment:
Are my eyes fooling me or is this method not used?

Reply:
i believe it is not used, I will take it out